pkg/deployments/tasks/20230109-gauge-adder-v3/test/task.fork.ts

```diff
@@ -0,0 +1,206 @@
+      await factoryTask.run({ force: true });
+      factory = await factoryTask.deployedInstance('LiquidityGaugeFactory');
+
+      expect(await factory.isGaugeFromFactory(ZERO_ADDRESS)).to.be.false;
+    });
+
+    // We need to grant permission to the admin to add the LiquidityGaugeFactory to the GaugeAdder, and also to add
+    // gauges from said factory to the GaugeController.
+    before('grant permissions', async () => {
+      const addFactoryAction = await actionId(gaugeAdder, 'addGaugeFactory');
+      const addGaugeAction = await actionId(gaugeAdder, 'addEthereumGauge');
+      const gaugeControllerAddGaugeAction = await actionId(
+        adaptorEntrypoint,
+        'add_gauge(address,int128)',
+        gaugeController.interface
+      );
+
+      await authorizer
+        .connect(daoMultisig)
+        .manageGranter(addFactoryAction, lmMultisig.address, TimelockAuthorizer.EVERYWHERE, true);
+      await authorizer
+        .connect(daoMultisig)
+        .manageGranter(addGaugeAction, lmMultisig.address, TimelockAuthorizer.EVERYWHERE, true);
+      await authorizer
+        .connect(daoMultisig)
+        .manageGranter(gaugeControllerAddGaugeAction, lmMultisig.address, TimelockAuthorizer.EVERYWHERE, true);
+
+      let tx = await authorizer
+        .connect(lmMultisig)
+        .grantPermissions([addFactoryAction], admin.address, [TimelockAuthorizer.EVERYWHERE]);
+      expectEvent.inReceipt(await tx.wait(), 'PermissionGranted', {
+        actionId: addFactoryAction,
+        account: admin.address,
+        where: TimelockAuthorizer.EVERYWHERE,
+      });
+
+      tx = await authorizer
+        .connect(lmMultisig)
+        .grantPermissions([addGaugeAction], admin.address, [TimelockAuthorizer.EVERYWHERE]);
+      expectEvent.inReceipt(await tx.wait(), 'PermissionGranted', {
+        actionId: addGaugeAction,
+        account: admin.address,
+        where: TimelockAuthorizer.EVERYWHERE,
+      });
+
+      // Granting `GaugeController#add_gauge` permissions to the entrypoint has a delay, so the permission needs
+      // to be scheduled and executed after the required time passes.
+      tx = await authorizer
+        .connect(lmMultisig)
+        .scheduleGrantPermission(gaugeControllerAddGaugeAction, gaugeAdder.address, TimelockAuthorizer.EVERYWHERE, []);
+      const event = expectEvent.inReceipt(await tx.wait(), 'ExecutionScheduled');
+      const scheduledExecutionId = event.args.scheduledExecutionId;
+
+      // The adder cannot add a gauge in the controller before the delay passes.
+      expect(
+        await authorizer.canPerform(gaugeControllerAddGaugeAction, gaugeAdder.address, TimelockAuthorizer.EVERYWHERE)
+      ).to.be.false;
+
+      await advanceTime(14 * DAY);
+      await authorizer.connect(lmMultisig).execute(scheduledExecutionId);
+
+      expect(await authorizer.canPerform(addFactoryAction, admin.address, TimelockAuthorizer.EVERYWHERE)).to.be.true;
```

Comment:
I think this action should also have a delay, but since delays didn't exist in the old authorizer it has to be migrated somewhere else.

See https://github.com/balancer-labs/balancer-v2-monorepo/blob/master/pkg/deployments/tasks/20221202-timelock-authorizer/input/mainnet.ts#L224 for ref.

Reply:
I think this is OOS for this PR. I think it would likely be done in a review of delays post migration.